src/probnum/randprocs/_random_process.py

```diff
@@ -0,0 +1,253 @@
+            output_dim)* -- Covariance of the process at ``x0`` and ``x1``. If
+            only ``x0`` is given the kernel matrix :math:`K=k(X_0, X_0)` is computed.
+        """  # pylint: disable=trailing-whitespace
+        raise NotImplementedError
+
+    def var(self, x: _InputType) -> _OutputType:
+        """Variance function.
+
+        Returns the variance function which is the value of the covariance or kernel
+        evaluated elementwise at ``x`` for each output dimension separately.
+
+        Parameters
+        ----------
+        x
+            *shape=(input_dim,) or (n, input_dim)* -- Input(s) to the variance function.
+
+        Returns
+        -------
+        var
+            *shape=(), (output_dim,) or (n, output_dim)* -- Variance of the
+            process at ``x``.
+        """
+        try:
+            cov = self.cov(x0=x)
+            if cov.ndim < 2:
+                return cov
+            elif cov.ndim == 2:
+                return np.diag(cov)
+            else:
+                return np.vstack(
+                    [np.diagonal(cov[:, :, i, i]) for i in range(self.output_dim)]
+                ).T
+        except NotImplementedError as exc:
+            raise NotImplementedError from exc
+
+    def std(self, x: _InputType) -> _OutputType:
+        """Standard deviation function.
+
+        Parameters
+        ----------
+        x
+            *shape=(input_dim,) or (n, input_dim)* -- Input(s) to the standard
+            deviation function.
+
+        Returns
+        -------
+        var
+            *shape=(), (output_dim,) or (n, output_dim)* -- Standard deviation of the
+            process at ``x``.
+        """
+        try:
+            return np.sqrt(self.var(x=x))
+        except NotImplementedError as exc:
+            raise NotImplementedError from exc
+
+    def sample(
+        self,
+        x: _InputType = None,
+        size: ShapeArgType = (),
+        random_state: RandomStateArgType = None,
+    ) -> Union[Callable[[_InputType], _OutputType], _OutputType]:
+        """Sample paths from the random process.
+
+        If no inputs are provided this function returns sample paths which are
+        callables, otherwise random variables corresponding to the input locations
+        are returned.
+
+        Parameters
+        ----------
+        x
+            *shape=(input_dim,) or (n, input_dim)* -- Evaluation input(s) of the
+            sample paths of the process. If ``None``, sample paths, i.e. callables are
+            returned.
+        size
+            Size of the sample.
+        random_state :
+            Random state of the random process. If None (or np.random), the global
+            :mod:`numpy.random` state is used. If integer, it is used to seed the local
+            :class:`~numpy.random.RandomState` instance.
+        """
+        if x is None:
+            return lambda x0: self._sample_at_input(
```

Comment:
I think this is a good idea, I am glad that you suggested it months back.
One tiny suggestion: could we maybe replace the `lambda` function with a full one here? Or is there any reason for an anonymous one other than that it is a one-liner?

Reply:
I thought about it some more and there is another point that may need to be taken into account:

The sample function as returned here is only valid if called exactly once; if called a second time, the resulting samples will be independent of the first samples (that were generated the first time), which they should not be if one thinks of this function returning a sample path that behaves like a function. I am fine with  this being written in the docstring, but do you perhaps have an idea how to solve this?

Reply:
Nice catch! You're right, that is unexpected behavior. When passing a `random_state` everything works as expected. I changed the function such that even when sampling paths, a `random_state` is derived from the global RNG. Now sample paths are deterministic functions.

Reply:
Setting the random state is a good starting point, but I meant something more subtle than that. 

For instance, consider the setting where this function is used to sample from a GP. If this sample is called at a set of inputs, it returns the correct joint samples at the inputs. If however, this function is called again at a different set of inputs, what should happen? (a) the samples should be independent of the previous samples, which happens if they are computed in the same way as the previous samples without taking those into account; (b) the samples should be conditioned on the previous samples (by, roughly speaking, conditioning the GP on hitting those previous samples and sampling from the respective posterior GP now). Do you understand what I mean? I find this hard to express in a comment section, please ping me offline if you want me to elaborate :) 

I think in light of "a sample from a random process being a function", (b) would be desired. This is annoying to implement though (Nina has done it before). The question could be avoided altogether if we would somehow enforce that this sampling function can only be called once, which is why I suggested the solution in my previous comment above. 

In any case, I don't think this problem is a neck breaker for this PR and we can merge anyway as soon as the rest is approved, but let's make an issue about this in that case.

Reply:
Okay, I understand now. This is indeed quite tricky. Allowing only a single evaluation of the sample path is equivalent to just offering sampling with input locations. I opted for removing the base implementation for sampling sample paths as functions and raising a `NotImplementedError` instead. That way if we come up with a strategy on how to do this efficiently in the future (or for a specific subclass) we can still implement it. I left the tests in and marked them as failing so that we can easily readd them in the future. Let me know whether that works for you, then I will merge!

Reply:
It does, thanks for taking care of this!